metadata-ingestion/src/datahub/utilities/source_helpers.py

```diff
@@ -80,9 +85,12 @@ def auto_stale_entity_removal(
     for wu in stream:
         urn = wu.get_urn()
 
-        entity_type = entity_type_fn(wu)
-        if entity_type is not None:
-            stale_entity_removal_handler.add_entity_to_state(entity_type, urn)
+        if wu.is_primary_source:
+            entity_type = entity_type_fn(wu)
+            if entity_type is not None:
+                stale_entity_removal_handler.add_entity_to_state(entity_type, urn)
+        else:
+            stale_entity_removal_handler.add_urn_to_skip(urn)
```

Comment:
qq - why do we also need to call this manually in the DBT source? it seems like just passing a workunit with primary false would exclude this thing?